File: src/main/java/com/github/zoewithabang/model/CommandInfo.java
package com.github.zoewithabang.model;

import java.util.Objects;

public class CommandInfo implements IModel
{
    private String command;
    private Boolean active;
    private Integer permissionRank;
    
    public CommandInfo() {}
    
    public CommandInfo(String command, Boolean active, Integer permissionRank)
    {
        this.command = command;
        this.active = active;
        this.permissionRank = permissionRank;
    }
    
    public String getCommand()
    {
        return command;
    }
    
    public void setCommand(String command)
    {
        this.command = command;
    }
    
    public Boolean getActive()
    {
        return active;
    }
    
    public void setActive(Boolean active)
    {
        this.active = active;
    }
    
    public Integer getPermissionRank()
    {
        return permissionRank;
    }
    
    public void setPermissionRank(Integer permissionRank)
    {
        this.permissionRank = permissionRank;
    }
    
    @Override
    public boolean equals(Object o)
    {
        if(this == o) return true;
        if(o == null || getClass() != o.getClass()) return false;
        CommandInfo that = (CommandInfo) o;
        return Objects.equals(command, that.command) &&
            Objects.equals(active, that.active) &&
            Objects.equals(permissionRank, that.permissionRank);
    }
    
    @Override
    public int hashCode()
    {
        return Objects.hash(command, active, permissionRank);
    }
    
    @Override
    public String toString()
    {
        return "CommandInfo{" +
            "command='" + command + '\'' +
            ", active=" + active +
            ", permissionRank=" + permissionRank +
            '}';
    }
}

File: src/main/kotlin/com/github/zoewithabang/model/CommandInfo.kt
package com.github.zoewithabang.model

class CommandInfo(command: String, active: Boolean, permissionRank: Int) {
    var command: String? = command
    var active: Boolean? = active
    var permissionRank: Int? = permissionRank

    override fun toString(): String {
        return "CommandInfo(command=$command, active=$active, permissionRank=$permissionRank)"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as CommandInfo

        if (command != other.command) return false
        if (active != other.active) return false
        if (permissionRank != other.permissionRank) return false

        return true
    }

    override fun hashCode(): Int {
        var result = command?.hashCode() ?: 0
        result = 31 * result + (active?.hashCode() ?: 0)
        result = 31 * result + (permissionRank ?: 0)
        return result
    }
}
